pkg/operator/ceph/cluster/mgr/mgr.go

```diff
@@ -591,3 +609,44 @@ func applyMonitoringLabels(c *Cluster, serviceMonitor *monitoringv1.ServiceMonit
 		}
 	}
 }
+
+func shouldRotateMgrKeys(c *clusterd.Context, clusterInfo *cephclient.ClusterInfo) (bool, error) {
+	cephObj := &cephv1.CephCluster{}
+	if err := c.Client.Get(clusterInfo.Context, clusterInfo.NamespacedName(), cephObj); err != nil {
+		return false, errors.Wrapf(err, "failed to get the cluster %v.", clusterInfo.NamespacedName())
+	}
+	desiredCephVersion := clusterInfo.CephVersion
+	// TODO: for rotation WithCephVersionUpdate fix this to have the right runningCephVersion and desiredCephVersion
+	runningCephVersion := clusterInfo.CephVersion
+
+	shouldRotateKeys, err := keyring.ShouldRotateCephxKeys(cephObj.Spec.Security.CephX.Daemon, runningCephVersion, desiredCephVersion, *cephObj.Status.Cephx.Mgr)
+	if err != nil {
+		return false, errors.Wrap(err, "failed to check if mgr daemon keys should be rotated or not")
+	}
+
+	return shouldRotateKeys, nil
+}
+
+func updateMgrCephxStatus(c *clusterd.Context, clusterInfo *cephclient.ClusterInfo, didRotate bool) (*cephv1.CephxStatus, error) {
+	updatedStatus := cephv1.CephxStatus{}
+	err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		cluster := &cephv1.CephCluster{}
+		if err := c.Client.Get(clusterInfo.Context, clusterInfo.NamespacedName(), cluster); err != nil {
+			return errors.Wrapf(err, "failed to get cluster %v to update the conditions.", clusterInfo.NamespacedName())
+		}
+		updatedStatus = keyring.UpdatedCephxStatus(didRotate, cluster.Spec.Security.CephX.Daemon, clusterInfo.CephVersion, *cluster.Status.Cephx.Mgr)
+		cluster.Status.Cephx.Mgr = &updatedStatus
+		logger.Debugf("updating mgr daemon cephx status to %+v", cluster.Status.Cephx.Mgr)
+		if err := reporting.UpdateStatus(c.Client, cluster); err != nil {
```

Comment:
do we need to use `Patch` instead of `update`? 
As we arent re-setting all the fields, and it can lead to setting other status fields as empty

Reply:
> it can lead to setting other status fields as empty

Can you elaborate a bit more on this?  Why would it lead to other status fields getting set as empty? The function is fetching the latest cephCluster instance and only trying to update the `Status.Cephx.Mgr`, keeping all the other status information as it is. But maybe I'm missing something!  @parth-gr